src/tokenizer.rs

```diff
@@ -43,11 +44,12 @@ lazy_static! {
     static ref PATTERN_CLOSE_PAREN: Regex = Regex::new(r"^(\))").unwrap();
 
     static ref PATTERN_WHITESPACE: Regex = Regex::new(r"^\s+").unwrap();
+    static ref PATTERN_CHARS_AFTER_NEWLINE: Regex = Regex::new(r"\n([^\n]+)$").unwrap();
```

Comment:
It's not pinned to the beginning, it's pinned to the end, so that a string that eats multiple newlines doesn't get false-positives. The `^` character is in a character class - it's inverting the class, not pinning the pattern to the beginning of the string.

Reply:
Oh, no, I was referring to the lack of a `^` at the beginning of the string, which *would* pin to the beginning.

I don't completely understand why it isn't pinned at both ends?

Reply:
Okay, I see now. I think I misunderstood the inputs and outputs at first.